runtime/common/addresslocation.go

```diff
@@ -80,12 +80,30 @@ func (l AddressLocation) MeteredID(memoryGauge MemoryGauge) LocationID {
 }
 
 func (l AddressLocation) TypeID(memoryGauge MemoryGauge, qualifiedIdentifier string) TypeID {
-	return NewMeteredTypeID(
-		memoryGauge,
-		AddressLocationPrefix,
-		l.Address.Hex(),
-		qualifiedIdentifier,
-	)
+	var i int
+
+	// AddressLocationPrefix '.' hex-encoded address '.' qualifiedIdentifier
```

Comment:
the TypeID impls for AddressLocation, ScriptLocation, and TransactionLocation seem almost identical save for the prefices and middle fields

Reply:
Agree it would be nice to refactor these to use a common helper if possible.

Reply:
Good idea, refactored in 36afaf932b06df966f5ac43d8429c6fee4ce1172